### PR TITLE
first pass at MYA site csv to json tool

### PR DESCRIPTION
**Pull request by jrowett**

### CODE REVIEW

File: data/CsvDataTool/SiteCsvReader.cs [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+                sites.Add(site);
+                report.Add(new SiteRowReportItem(index, site.Name, true, ""));
+                index++;
+            }       
+        }
+
+        return (sites.ToArray(), report.ToArray());
+    }
+
+    private TextReader GetCsvStreamReader()
```
Comment (vicr1) [inline data/CsvDataTool/SiteCsvReader.cs:75]: This could be replaced by using the Lazy pattern

File: data/CsvDataTool/SiteJsonWriter.cs
Comment (vicr1) [inline data/CsvDataTool/SiteJsonWriter.cs:1]: Seems like a potential gotcha having this duplicated

File: data/CsvDataTool/SiteMap.cs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            new Location(
+                "Point",
+                [x.Row.GetField<double>("Longitude"), x.Row.GetField<double>("Latitude")]
+                ));
+        Map(m => m.IntegratedCareBoard).Name("ICB");
+        Map(m => m.Region).Name("Region");
+    }
+}
+
+public record struct Site(
```
Comment (vicr1) [inline data/CsvDataTool/SiteMap.cs:24]: Can we reference the existing types

**Review by vicr1 — CHANGES_REQUESTED** [review]
Should be marked as draft as this is not a complete piece of work

File: data/CsvDataTool/SiteCsvReader.cs [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+            Quote = '"'
+        };
+
+        using (var reader = GetCsvStreamReader())
+        using (var csv = new CsvReader(reader, config))
+        {
+            csv.Context.RegisterClassMap<SiteMap>();
+
+            var imported = csv.GetRecords<Site>();
+            foreach(var site in imported)
```
[the remark this answers is not in the document]
Comment (jrowett) [reply data/CsvDataTool/SiteCsvReader.cs:64]: i don't see how this increments the value of index

File: data/CsvDataTool/SiteMap.cs (hunk 2 above)
Comment (jrowett) [reply data/CsvDataTool/SiteMap.cs:24]: the CSV library has slightly different requirements for the DTOs, they need default constructors. this was my original approach until i encountered this gotcha in the library

File: data/CsvDataTool/SiteJsonWriter.cs
Comment (jrowett) [reply data/CsvDataTool/SiteJsonWriter.cs:1]: puts s dependency from this tool onto the API function app - we could cleave off json tasks to their own library i guess

**Review by vicr1 — APPROVED** [review]

**Pull request merged**